semgrep-core/src/engine/Match_rules.ml

```diff
@@ -399,7 +399,7 @@ type ('target_content, 'xpattern) xpattern_matcher = {
     'target_content ->
     filename ->
     'xpattern ->
-    (Parse_info.token_location * MV.bindings) list;
+    (Parse_info.token_location * Parse_info.token_location * MV.bindings) list;
```

Comment:
Maybe a comment explaining why this is needed?

Reply:
I mean why we need two `token_location` instead of one.

Reply:
yep, will do.